api/snapshot.go

```diff
@@ -0,0 +1,691 @@
+		url := fmt.Sprintf("http://%s:%d%s/report/snapshot/status", master.IP, config.FlagPort, BaseRoute)
+		body, _, err := DCOSTools.Get(url, time.Duration(time.Second*3))
+		if err = json.Unmarshal(body, &status); err != nil {
+			log.Error(err)
+			continue
+		}
+		statuses[master.IP] = status
+	}
+	if len(statuses) == 0 {
+		return statuses, errors.New("Could not determine wheather the snapshot job is running or not.")
+	}
+	return statuses, nil
+}
+
+// get a status report for a localhost
+func (j *SnapshotJob) getStatus(config *Config) snapshotReportStatus {
+	// use a temp var `used`, since disk.Usage panics if partition does not exist.
+	var used float64
+	usageStat, err := disk.Usage(config.FlagSnapshotDir)
+	if err == nil {
+		used = usageStat.UsedPercent
+	} else {
+		log.Errorf("Could not get a disk usage: %s", config.FlagSnapshotDir)
+	}
+	return snapshotReportStatus{
+		Running:          j.Running,
+		Status:           j.Status,
+		Errors:           j.Errors,
+		LastSnapshotPath: j.LastSnapshotPath,
+		JobStarted:       j.JobStarted.String(),
+		JobEnded:         j.JobEnded.String(),
+		JobDuration:      j.JobDuration.String(),
+
+		SnapshotBaseDir:                       config.FlagSnapshotDir,
+		SnapshotJobTimeoutMin:                 config.FlagSnapshotJobTimeoutMinutes,
+		SnapshotJobGetSingleURLTimeoutMinutes: config.FlagSnapshotJobGetSingleURLTimeoutMinutes,
+		SnapshotUnitsLogsSinceHours:           config.FlagSnapshotUnitsLogsSinceHours,
+		CommandExecTimeoutSec:                 config.FlagCommandExecTimeoutSec,
+
+		DiskUsedPercent: used,
+	}
+}
+
+// fetch an HTTP endpoint and append the output to a zip file.
+func (j *SnapshotJob) getHTTPAddToZip(node Node, endpoints map[string]string, folder string, zipWriter *zip.Writer,
+	summaryReport *bytes.Buffer, config *Config, DCOSTools DCOSHelper) error {
+	for fileName, httpEndpoint := range endpoints {
+		fullURL := "http://" + node.IP + httpEndpoint
+		j.Status = "GET " + fullURL
+		log.Debug(j.Status)
+		timeout := time.Duration(time.Minute * time.Duration(config.FlagSnapshotJobGetSingleURLTimeoutMinutes))
+		request, err := http.NewRequest("GET", fullURL, nil)
+		if err != nil {
+			j.Errors = append(j.Errors, err.Error())
+			log.Error(err)
+			updateSummaryReport(fmt.Sprintf("could not create request for url: %s", fullURL), node, err.Error(), summaryReport)
+			continue
+		}
+		request.Header.Add("Accept-Encoding", "gzip")
+		resp, err := DCOSTools.HTTPRequest(request, timeout)
+		if err != nil {
+			j.Errors = append(j.Errors, err.Error())
+			log.Errorf("Could not fetch url: %s", fullURL)
+			log.Error(err)
+			updateSummaryReport(fmt.Sprintf("could not fetch url: %s", fullURL), node, err.Error(), summaryReport)
+			continue
+		}
+		if resp.Header.Get("Content-Encoding") == "gzip" {
+			fileName += ".gz"
+		}
+
+		// put all logs in a `ip_role` folder
+		zipFile, err := zipWriter.Create(filepath.Join(node.IP+"_"+node.Role, fileName))
+		if err != nil {
+			resp.Body.Close()
```

Comment:
`resp.Body.Close()` should drop in on line 398 as `defer resp.Body.Close()`. Then you can remove it here and below the if on 414.